lib/ui/painting.dart

```diff
@@ -3676,9 +3691,13 @@ class ImageShader extends Shader {
   /// tile. The second and third arguments specify the [TileMode] for the x
   /// direction and y direction respectively. The fourth argument gives the
   /// matrix to apply to the effect. All the arguments are required and must not
-  /// be null.
+  /// be null, except for [filterQuality]. If [filterQuality] is not specified
+  /// at construction time it will be deduced from the environment where it is used,
+  /// such as from [Paint.filterQuality].
   @pragma('vm:entry-point')
-  ImageShader(Image image, TileMode tmx, TileMode tmy, Float64List matrix4) :
+  ImageShader(Image image, TileMode tmx, TileMode tmy, Float64List matrix4, {
+    FilterQuality? filterQuality,
```

Comment:
Does this need to be nullable, or could it be non-nullable with a default?

Reply:
Currently ImageShaders take their quality from the associated paint at the time they are used in a paint call. We need to preserve that notion to avoid breaking backwards compatibility.